leapp/actors/__init__.py

```diff
@@ -13,73 +13,130 @@
 
     def get_folder_path(self, name):
+        """
+        Displays first folder path within actor.
+
+        :param name: Name of the folder
+        :type name: str
+        :return: Found folder path
+        :rtype: str or None
+        """
         for path in self.files_paths:
             path = os.path.join(path, name)
             if os.path.isdir(path):
                 return path
         return None
 
     def get_file_path(self, name):
+        """
+        Displays first file path within actor and common actors file paths.
+
+        :param name: Name of the file
+        :type name: str
+        :return: Found file path
+        :rtype: str or None
+        """
         for path in self.files_paths:
             path = os.path.join(path, name)
             if os.path.isfile(path):
                 return path
         return None
 
     def run(self, *args):
+        """ Runs actor calling method :py:func:`process`. """
         os.environ['LEAPP_CURRENT_ACTOR'] = self.name
         try:
             self.process(*args)
         finally:
             os.environ.pop('LEAPP_CURRENT_ACTOR', None)
 
     def process(self, *args, **kwargs):
+        """ Main processing method (in inherited actors, the function needs to be defined to be able process)"""
         raise NotImplementedError()
 
-    def produce(self, *args):
+    def produce(self, *models):
+        """
+        After running :py:func:`process`, it prepares processed data - results to next processing in workflow.
+
+        :param models: what data models is going to be produced
+        :type models: tuple of derived classes from :py:class:`leapp.models.Model`
+        """
         if self._messaging:
-            for arg in args:
-                if isinstance(arg, getattr(self.__class__, 'produces')):
-                    message_data = json.dumps(arg.dump(), sort_keys=True)
+            for model in models:
+                if isinstance(model, getattr(self.__class__, 'produces')):
+                    message_data = json.dumps(model.dump(), sort_keys=True)
                     message_hash = hashlib.sha256(message_data).hexdigest()
-                    self._messaging.produce(arg.topic.name, {
-                        'type': arg.__class__.__name__,
+                    self._messaging.produce(model.topic.name, {
+                        'type': model.__class__.__name__,
                         'actor': self.name,
-                        'topic': arg.topic.name,
+                        'topic': model.topic.name,
                         'stamp': datetime.datetime.utcnow().isoformat() + 'Z',
                         'message': {
                             'data': message_data,
                             'hash': message_hash
                         }
                     })
 
-    def consume(self, *types):
+    def consume(self, *models):
+        """
+        Load any given model data types to be able to process them.
```

Comment:
Retrieve messages specified in the actors :py:attr:`~consumes` attribute and can be filter the message types by `models`